blueprints/incredible-squaring/src/main.rs

```diff
@@ -17,35 +21,46 @@ async fn main() -> Result<()> {
     logger.init();
 
     let env = gadget_sdk::env::load()?;
-    let keystore = env.keystore()?;
-    let client = subxt::OnlineClient::from_url(&env.tangle_rpc_endpoint).await?;
+    let client = env.client().await?;
+    let signer = env.first_signer()?;
+    if env.should_run_registration() {
```

Comment:
It is up to the blueprint developer to implement their custom registration code, this the whole idea, so no, this part cannot be abstracted, since each blueprint will require different registration params, if any at all.